### PR TITLE
Add input.error and input.errorMessage props to AutocompleteInput

### PR DESCRIPTION
**Pull request by estacioneto**
#### What is the purpose of this pull request?

As the title says...

#### What problem is this solving?

There was no error state on `AutocompleteInput` component. [Discussion on Slack](https://vtex.slack.com/archives/C1MCRG7CK/p1595862494183500)

#### Screenshots or example usage

image

#### Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/AutocompleteInput/SearchInput/index.tsx [hunk 1]
```diff
@@ -26,6 +26,10 @@ const propTypes = {
   disabled: PropTypes.bool,
   /** Determine the search bar size */
   size: PropTypes.oneOf(['small', 'regular', 'large']),
+  /** Determine if the input and button should be styled with error borders */
+  error: PropTypes.bool,
+  /** The error message to be desiplayed below the input field */
```
Comment (vlaux) [inline react/components/AutocompleteInput/SearchInput/index.tsx:31]:
```suggestion
  /** The error message to be displayed below the input field */
```

File: react/components/AutocompleteInput/index.tsx [hunk 2]
```diff
@@ -26,6 +26,10 @@ const propTypes = {
     value: PropTypes.string,
     /** Determine if the input and button should be disabled */
     disabled: PropTypes.bool,
+    /** Determine if the input and button should be styled with error borders */
+    error: PropTypes.bool,
+    /** The error message to be desiplayed below the input field */
```
Comment (vlaux) [inline react/components/AutocompleteInput/index.tsx:31]:
```suggestion
    /** The error message to be displayed below the input field */
```

Comment (estacioneto) [reply react/components/AutocompleteInput/index.tsx:31]: Nice! Thanks 🚀

**Comment by estacioneto** [comment]
Thank you @jonathasbsouza! You're completely right about the error message 💯 

About the error state programmatically changing the button's color on input's focus, I'm afraid we can't do that (ring a bell? 🔔 ). That's because the class used on the input is `hover-b--danger` and this color is only present on hover (you can see this by hovering the search button).
Do you have a suggestion? Do you know a Tachyons token with the `hover-b--danger` color that can be used here?

**Pull request merged**